homeassistant/components/recorder/statistics.py

```diff
@@ -343,12 +343,13 @@ def _delete_duplicates_from_table(
         if not duplicate_ids:
             break
         all_non_identical_duplicates.extend(non_identical_duplicates)
-        deleted_rows = (
-            session.query(table)
-            .filter(table.id.in_(duplicate_ids))
-            .delete(synchronize_session=False)
-        )
-        total_deleted_rows += deleted_rows
+        for i in range(0, len(duplicate_ids), MAX_ROWS_TO_PURGE):
+            deleted_rows = (
```

Comment:
From #65682 it sounds like the problem is the finding of duplicates, not the removal?

Reply:
This patch means we only need to find the duplicates a limited number of times, instead of thousands of times when there are many duplicates.